pkg/server/image_status.go

```diff
@@ -49,6 +70,59 @@ func (c *criContainerdService) ImageStatus(ctx context.Context, r *runtime.Image
 	}
 	runtimeImage.Username = username
 
-	// TODO(mikebrow): write a ImageMetadata to runtime.Image converter
-	return &runtime.ImageStatusResponse{Image: runtimeImage}, nil
+	return runtimeImage
+}
+
+// TODO (mikebrow): discuss moving this struct and / or constants for info map for some or all of these fields to CRI
+type verboseImageInfo struct {
+	Config             *imagespec.ImageConfig `json:"config"`
+	ConfigDescriptor   imagespec.Descriptor   `json:"configDescriptor"`
+	ManifestDescriptor imagespec.Descriptor   `json:"manifestDescriptor"`
+	LayerInfo          []content.Info         `json:"layerInfo"`
+}
+
+// toCRIImageInfo converts internal image object information to CRI image status response info map.
+func (c *criContainerdService) toCRIImageInfo(ctx context.Context, image *imagestore.Image, verbose bool) (map[string]string, error) {
+	if !verbose {
+		return nil, nil
+	}
+
+	info := make(map[string]string)
+	i := image.Image
+	descriptor, err := i.Config(ctx)
+	if err != nil {
+		glog.Errorf("Failed to get image config %q: %v", image.ID, err)
+	} // fallthrough
+
+	targetDescriptor := i.Target()
+	var dia []content.Info
+	digests, err := i.RootFS(ctx)
+	if err != nil {
+		glog.Errorf("Failed to get target digests %q: %v", i.Name(), err)
+	} else {
+		dia = make([]content.Info, len(digests))
+		for i, d := range digests {
```

Comment:
This is wrong. `RootFS` digest is uncompressed digest, content is compressed.
You won't be able to get things from content store info:
```
{
  "status": {
    "id": "sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27",
    "repoTags": [
      "docker.io/library/busybox:1.26"
    ],
    "repoDigests": [
      "docker.io/library/busybox@sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642"
    ],
    "size": "701277",
    "username": ""
  },
  "info": {
    "config": {
      "Env": [
        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      ],
      "Cmd": [
        "sh"
      ]
    },
    "configDescriptor": {
      "mediaType": "application/vnd.docker.container.image.v1+json",
      "digest": "sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27",
      "size": 1507
    },
    "manifestDescriptor": {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642",
      "size": 527
    },
    "layerInfo": [
      {
        "Digest": "",
        "Size": 0,
        "CreatedAt": "0001-01-01T00:00:00Z",
        "UpdatedAt": "0001-01-01T00:00:00Z",
        "Labels": null
      }
    ]
  }
}
```

We may want to include the whole image spec, which has already contained everything.
We also need ChainID here, because that's how we find the corresponding snapshot in containerd.

Reply:
Agree we should omit, if content isn't accessible.
Agree, we may want to include the whole imagespec.Image.
Agree, to add chainID.

Reply:
@mikebrow You could change `imagestore.Image.Config` to `imagestore.Image.ImageSpec` and change all the other referencing place.

So that you don't need to read that from containerd again.

Reply:
@Random-Liu added a commit to place the full OCI image spec into the imagestore.. Debugging it now.